### PR TITLE
Added support for Document Types (view, edit, filterby)

### PR DESCRIPTION
**Pull request by knom**
@bauerj 
I'm heavily using Document Types (letter, invoice, ...) in Paperless-NGX.

So I thought it would be great to have them in the app as well.

Not sure how this would work correctly with paperless NG?

### CODE REVIEW

File: lib/api.dart [hunk 1]
```diff
@@ -167,15 +196,24 @@ class _Converter<T> implements JsonConverter<T?, Object?> {
...
     if (json is Map<String, dynamic> && json.containsKey('colour')) {
       return OgTag.fromJson(json) as T;
     }
     if (json is Map<String, dynamic> && json.containsKey('color')) {
       return Tag.fromJson(json) as T;
     }
-    if (json is Map<String, dynamic> && json.containsKey('name')) {
+    if (json is Map<String, dynamic> &&
+        json.containsKey('name') &&
+        isType<T, Correspondent?>()) {
```
Comment (bauerj) [inline lib/api.dart:209]: That's clever!

**Comment by bauerj** [comment]
Thanks a lot for your PR!

The code looks good on a quick glance. I'm going to test this on Monday and then get back to you with more feedback.

**Comment by knom** [comment]
Great, thank you!
I'm not using F-Droid, but the regular Google store.

I realized since November no new release has been published there.. 
So it would be great if you could publish a new release there, once we're done with the PR.

Thanks.

**Review by bauerj — APPROVED** [review]

**Comment by bauerj** [comment]
The new release is now out. Sorry for the delay!

**Pull request merged**